public/components/reindex.react.js

```diff
@@ -52,18 +62,30 @@ export default class ReindexComponent extends React.Component {
                         </Col>
                         <Col xs={6} md={6}>
                             <Panel header="Details">
-                                <ContentSource key={this.state.contentSource.id}
-                                               id={this.state.contentSource.id}
-                                               description={this.state.contentSource.description}
-                                               appName={this.state.contentSource.appName}
-                                               reindexEndpoint={this.state.contentSource.reindexEndpoint}/>
+                                {this.state.editModeOn ?
+                                    <ContentSourceEdit key={this.state.contentSource.id}
```

Comment:
I think it's completely reasonable for a component named <ContentSourceEdit /> to accept a contentSource rather than having to pass through the individual fields. Would make this a little easier to read